Tests/DataParser/test_dataParser.py

```diff
@@ -0,0 +1,116 @@
+import os
+import json
+import xmltodict
+import zipfile
+import pandas as pd
+from pathlib import Path
+import pytest
+import sys
+sys.path.append("../../DataParser")
+import filecmp
+import dataParser
+
+testFiles = "./testFiles"
+truth_folder = "./testFiles/truth_data"
+
+__TESTS__ = 4 
+
+def build_testdata():
+	"""
+	This method builds the gold directory structure for the files added
+	in the test folder of the data path. This created directory structure 
+	will be used while comparing the test.
+	"""
+	os.system("rm -rf ./testFiles/truth_data/*")
+	for i in range(1,__TESTS__+1):
+		print("Preparing Test :",i)
+		os.system("rm -rf ./testFiles/test"+str(i)+"/generated")
+		dataParser.prepare_folder("./testFiles/test"+str(i))
+		os.system("cp -rf ./testFiles/test"+str(i)+" ./testFiles/truth_data")
+		os.system("rm -rf ./testFiles/test"+str(i)+"/generated")
+
+
+# --build-test in command line argument to generate the gold test data
+if "--build" in sys.argv:
+	build_testdata()
+
+
+def are_dir_trees_equal(dir1, dir2):
+	"""
+	Compare two directories recursively. Files in each directory are
+	assumed to be equal if their names and contents are equal.
+
+	@param dir1: First directory path
+	@param dir2: Second directory path
+
+	@return: True if the directory trees are the same and 
+		there were no errors while accessing the directories or files, 
+		False otherwise.
+	"""
+	dirs_cmp = filecmp.dircmp(dir1, dir2)
+	if len(dirs_cmp.left_only)>0 or len(dirs_cmp.right_only)>0 or \
+		len(dirs_cmp.funny_files)>0:
+		return False
+
+	(_, mismatch, errors) =  filecmp.cmpfiles(
+		dir1, dir2, dirs_cmp.common_files, shallow=False)
+	
+	if len(mismatch)>0 or len(errors)>0:
+		return False
+	
+	for common_dir in dirs_cmp.common_dirs:
+		new_dir1 = os.path.join(dir1, common_dir)
+		new_dir2 = os.path.join(dir2, common_dir)
+		if not are_dir_trees_equal(new_dir1, new_dir2):
+			return False
+	
+	return True
+
+
+# The pytest testing function
+
+# @pytest.mark.set1
+def test_method1():
```

Comment:
The test name should be reflective of what is getting tested. Please rename all the test method names accordingly. Example here: test_word_files_conversion